src/main/native/cpp/analysis/AnalysisManager.cpp

```diff
@@ -158,68 +175,75 @@ static void PrepareGeneralData(const wpi::json& json,
-      data["fast-backward"], settings.windowSize);
-
-  // Compute acceleration on median filtered data sets.
-  auto slowForward = ComputeAcceleration<4, kVoltageCol, kPosCol, kVelCol>(
-      sysid::ApplyMedianFilter<4, kVelCol>(data["slow-forward"],
-                                           settings.windowSize),
-      settings.windowSize);
-  auto slowBackward = ComputeAcceleration<4, kVoltageCol, kPosCol, kVelCol>(
-      sysid::ApplyMedianFilter<4, kVelCol>(data["slow-backward"],
-                                           settings.windowSize),
-      settings.windowSize);
-  auto fastForward = ComputeAcceleration<4, kVoltageCol, kPosCol, kVelCol>(
-      sysid::ApplyMedianFilter<4, kVelCol>(data["fast-forward"],
-                                           settings.windowSize),
-      settings.windowSize);
-  auto fastBackward = ComputeAcceleration<4, kVoltageCol, kPosCol, kVelCol>(
-      sysid::ApplyMedianFilter<4, kVelCol>(data["fast-backward"],
-                                           settings.windowSize),
-      settings.windowSize);
-
-  // Calculate cosine of position data.
-  CalculateCosine(&slowForward, unit);
-  CalculateCosine(&slowBackward, unit);
-  CalculateCosine(&fastForward, unit);
-  CalculateCosine(&fastBackward, unit);
+  ApplyToData<std::vector<Data>>(
+      data,
+      [&](wpi::StringRef key) {
+        sysid::TrimQuasistaticData<4, kVoltageCol, kVelCol>(
+            &data[key], settings.motionThreshold);
+      },
+      [](wpi::StringRef key) { return key.contains("slow"); });
+
+  // Apply Median filter
+  ApplyToData<std::vector<Data>>(
+      data,
+      [&](wpi::StringRef key) {
+        sysid::ApplyMedianFilter<4, kVelCol>(data[key], settings.windowSize);
+      },
+      [](wpi::StringRef key) { return !key.startswith("raw"); });
+
+  // Get accel
+  ApplyToData<std::vector<Data>>(data, [&](wpi::StringRef key) {
+    preparedData[key] = ComputeAcceleration<4, kVoltageCol, kPosCol, kVelCol>(
+        data[key], settings.windowSize);
+  });
+
+  // Calculate cosine of position data for filtered data
+  ApplyToData<std::vector<PreparedData>>(
+      preparedData,
+      [&](wpi::StringRef key) { CalculateCosine(&preparedData[key], unit); },
+      [](wpi::StringRef key) { return !key.startswith("raw"); });
 
   // Find the maximum Step Test Duration
   maxStepTime = GetMaxTime<4>(data, kTimeCol);
 
   // Trim the raw step voltage data.
-  auto tempTime = units::second_t{
-      0.0};  // Raw data shouldn't be used to calculate mininum step test time
-  sysid::TrimStepVoltageData(&rawFastForward, settings, tempTime, maxStepTime);
-  sysid::TrimStepVoltageData(&rawFastBackward, settings, tempTime, maxStepTime);
-
-  // Trim the step voltage data.
-  sysid::TrimStepVoltageData(&fastForward, settings, minStepTime, maxStepTime);
-  sysid::TrimStepVoltageData(&fastBackward, settings, minStepTime, maxStepTime);
+  auto tempTime = units::second_t{0.0};
```

Comment:
This variable needs a better name.
```suggestion
  auto tempTime = 0_s;
```

Reply:
Maybe `dummyTime` ?

Reply:
How about `kInvalidTime`?

Reply:
We need it because the step trim is supposed to modify the minTime in the settings, but I didn't want the raw data to influence that calculation so instead the function modifies tempTime. Should I just pass a zero into the function instead?